src/Microsoft.AspNet.Mvc.TagHelpers/InputTagHelper.cs

```diff
@@ -121,93 +122,79 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                 output.CopyHtmlAttribute(nameof(Value), context);
             }
 
-            if (For == null)
+            // Note null or empty For.Name is allowed because TemplateInfo.HtmlFieldPrefix may be sufficient.
+            // IHtmlGenerator will enforce name requirements.
+            var metadata = For.Metadata;
+            var modelExplorer = For.ModelExplorer;
+            if (metadata == null)
```

Comment:
This was previous code, the diff is just misleading. Since we require the `asp-for` attribute to be provided there was no longer a need to check if it was provided.